models/migrations/v1_20/v259.go

```diff
@@ -0,0 +1,387 @@
+	AccessTokenScopeReadNotification:   accessTokenScopeReadNotificationBits,
+	AccessTokenScopeWriteNotification:  accessTokenScopeWriteNotificationBits,
+	AccessTokenScopeDeleteNotification: accessTokenScopeDeleteNotificationBits,
+	AccessTokenScopeReadOrganization:   accessTokenScopeReadOrganizationBits,
+	AccessTokenScopeWriteOrganization:  accessTokenScopeWriteOrganizationBits,
+	AccessTokenScopeDeleteOrganization: accessTokenScopeDeleteOrganizationBits,
+	AccessTokenScopeReadPackage:        accessTokenScopeReadPackageBits,
+	AccessTokenScopeWritePackage:       accessTokenScopeWritePackageBits,
+	AccessTokenScopeDeletePackage:      accessTokenScopeDeletePackageBits,
+	AccessTokenScopeReadIssue:          accessTokenScopeReadIssueBits,
+	AccessTokenScopeWriteIssue:         accessTokenScopeWriteIssueBits,
+	AccessTokenScopeDeleteIssue:        accessTokenScopeDeleteIssueBits,
+	AccessTokenScopeReadRepository:     accessTokenScopeReadRepositoryBits,
+	AccessTokenScopeWriteRepository:    accessTokenScopeWriteRepositoryBits,
+	AccessTokenScopeDeleteRepository:   accessTokenScopeDeleteRepositoryBits,
+	AccessTokenScopeReadUser:           accessTokenScopeReadUserBits,
+	AccessTokenScopeWriteUser:          accessTokenScopeWriteUserBits,
+	AccessTokenScopeDeleteUser:         accessTokenScopeDeleteUserBits,
+}
+
+// hasScope returns true if the string has the given scope
+func (bitmap accessTokenScopeBitmap) hasScope(scope AccessTokenScope) (bool, error) {
+	expectedBits, ok := allAccessTokenScopeBits[scope]
+	if !ok {
+		return false, fmt.Errorf("invalid access token scope: %s", scope)
+	}
+
+	return bitmap&expectedBits == expectedBits, nil
+}
+
+// toScope returns a normalized scope string without any duplicates.
+func (bitmap accessTokenScopeBitmap) toScope(unknownScopes *[]unknownAccessTokenScope) AccessTokenScope {
+	var scopes []string
+
+	// Preserve unknown scopes, and put them at the beginning so that it's clear
+	// when debugging.
+	if unknownScopes != nil {
+		for _, unknownScope := range *unknownScopes {
+			scopes = append(scopes, string(unknownScope))
+		}
+	}
+
+	// iterate over all scopes, and reconstruct the bitmap
+	// if the reconstructed bitmap doesn't change, then the scope is already included
+	var reconstruct accessTokenScopeBitmap
+
+	for _, singleScope := range allAccessTokenScopes {
+		// no need for error checking here, since we know the scope is valid
+		if ok, _ := bitmap.hasScope(singleScope); ok {
+			current := reconstruct | allAccessTokenScopeBits[singleScope]
+			if current == reconstruct {
+				continue
+			}
+
+			reconstruct = current
+			scopes = append(scopes, string(singleScope))
+		}
+	}
+
+	scope := AccessTokenScope(strings.Join(scopes, ","))
+	scope = AccessTokenScope(strings.ReplaceAll(
+		string(scope),
+		"delete:activitypub,delete:admin,delete:misc,delete:notification,delete:organization,delete:package,delete:issue,delete:repository,delete:user",
+		"all",
+	))
```

Comment:
Could the `accessTokenScopeBitmap` be put in the `modules` package (instead of the `models` package) ?

Then the code could be shared by this migration package, no need to duplicate.